### PR TITLE
T2S public bool is speaking

### PR DESCRIPTION
**Pull request by david-niggli**
TextToSpeechManager can now be queried whether or not it is currently speaking text.

### CODE REVIEW

**Comment by NeerajW** [comment]
Looks good.

**Comment by jwittner** [comment]
I agree, was hoping @jbienz would comment. Wondering if there's a more full proof way to detect if the audio was still playing than testing the duration.

**Comment by NeerajW** [comment]
I wonder if taking a leaf from this PR makes sense: https://github.com/Microsoft/HoloToolkit-Unity/pull/145

**Comment by keveleigh** [comment]
Since, with the way the script is set up, we only have one AudioSource associated with each TextToSpeechManager, we could just pass the AudioSource's isPlaying through.

If we expect the AudioSource to be reused for multiple things, we could set an isSpeaking flag in TextToSpeechManager when SpeakText/SSML() is called, then query audioSource.isPlaying in Update or a coroutine and reset the new flag once the AudioSource stops playing.

This would fail if the AudioSource is instantly given something new to play the exact frame it ends, or if the clip is changed mid-playback, and thus plays longer than just the TTS clip without stopping.

**Comment by david-niggli** [comment]
@keveleigh : Using the AudioSource's isPlaying was my first attempt too, but my character did not open his mouth. Maybe I had some other bug, you tell me...
Reading unity's documentation on AudioSource.isPlaying, it is apparently just !isPaused though.

**Comment by keveleigh** [comment]
From my tests just now, AudioSource.isPlaying will return true any time it's actively playing a clip. I think the note on Pause() is just to point out that a paused state is considered a "not playing" state, even though it might be in the middle of a clip.

I just downloaded your fork and replaced the code in IsSpeaking() with `return audioSource.isPlaying;`, and the cubes changed colors properly, from red to green and back to red. I'm not sure if that means there's a bug in your other app or not.

**Comment by david-niggli** [comment]
Well then I guess what I did is not as useful after all...

**Comment by jbienzms** [comment]
Sorry guys, I'm all over the place right now.

I went back and looked at the code I wrote for the Holo Voice Memo sample I show here:

https://youtu.be/xWlQc2vaumg

And here's how I implemented IsPlaying:

```
public bool IsPlaying
{
    get
    {
        return ((audioSource != null) && (audioSource.isPlaying));
    }
}
```

If you watch the video you can see the state doesn't seem to update immediately when switching from Playing to Stopped at the end of the clip. But that may be the way I was writing the code to update the visual state. Whenever Play would get called on my "Recording" class I would call:

`Invoke("EndOfClip", audioClip.length);`

And in EndOfClip I would look at IsPlaying and update the state accordingly. But that was never working right. It seemed that IsPlaying was always true. So I ended up doing this instead:

`Invoke("EndOfClip", audioClip.length + 0.5f);`

And that works but it always felt like it took longer than it should to update. So I could never tell if reading AudioClip.isPlaying was really the right way to go, but based on @keveleigh research above, it seems like that should be the right way to do it. What I should have probably done instead is check isPlaying on Update, and then only when the state actually changed from the last time I checked I could update the visuals.

In short, I think I agree that AudioClip.isPlaying is probably just about as good as having TTS.IsSpeaking. AudioClip would also theoretically account for changes in its own playback rate, etc.

**Comment by david-niggli** [comment]
Finally =)
Thanks for YOUR feedback, I think this can be closed then.
I have also successfully tested AudioClip.isPlaying by now and I can't explain why it didn't work the first time. I was wondering why you though it would be a good idea.
